exchanges/bitmex/bitmex_websocket.go

```diff
@@ -151,46 +128,47 @@ func (b *Bitmex) wsReadData() {
 }
 
 func (b *Bitmex) wsHandleData(respRaw []byte) error {
-	var err error
-	msg, _, _, err := jsonparser.Get(respRaw, "[3]")
-	if err != nil {
-		return fmt.Errorf("unknown message format: %s", respRaw)
-	}
 	// We don't need to know about errors, since we're looking optimistically into the json
-	op, _ := jsonparser.GetString(msg, "request", "op")
-	errMsg, _ := jsonparser.GetString(msg, "error")
-	success, _ := jsonparser.GetBoolean(msg, "success")
-	version, _ := jsonparser.GetString(msg, "version")
+	op, _ := jsonparser.GetString(respRaw, "request", "op")
+	errMsg, _ := jsonparser.GetString(respRaw, "error")
+	success, _ := jsonparser.GetBoolean(respRaw, "success")
+	version, _ := jsonparser.GetString(respRaw, "version")
 	switch {
 	case version != "":
-		op = "open"
-		fallthrough
-	case errMsg != "", success:
-		streamID, e2 := jsonparser.GetString(respRaw, "[1]")
-		if e2 != nil {
-			return fmt.Errorf("%w parsing stream", e2)
+		var welcomeResp WebsocketWelcome
+		if err := json.Unmarshal(respRaw, &welcomeResp); err != nil {
+			return err
+		}
+
+		if b.Verbose {
+			log.Debugf(log.ExchangeSys, "%s successfully connected to websocket API at time: %s Limit: %d", b.Name, welcomeResp.Timestamp, welcomeResp.Limit.Remaining)
 		}
-		err = b.Websocket.Match.RequireMatchWithData(op+":"+streamID, msg)
+		return nil
+	case errMsg != "", success:
+		req, _, _, err := jsonparser.Get(respRaw, "request")
```

Comment:
```suggestion
		var req any
		if op == "authKeyExpires" {
			req = op
		} else {
			reqBytes, _, _, err := jsonparser.Get(respRaw, "request")
			if err != nil {
				return err
			}
			req = string(reqBytes)
		}
		if err := b.Websocket.Match.RequireMatchWithData(req, respRaw); err != nil {
			return fmt.Errorf("%w: %s", err, op)
		}
		return nil
```

Since we explicitly set the match sig to be `authKeyExpires` and therefore need this logic to match it correctly

Reply:
Agreed. At first I thought we would treat it the same, but the Arguments are obviously sensitive.
Thanks for the suggestion!

Fixed https://github.com/gbjk/gocryptotrader/commit/91cf3d7f7f